### DISCUSSION TITLE
storing user images in public storage bucket and its implications

### DISCUSSION THREAD

**Discussion by wkev-n** [supabase/supabase#13940] [Questions] [answered]
I don't have direct experience with using s3, so I'd like to understand the implications of making a storage bucket public.

1. Does making a storage bucket public in supabase also mean the objects in s3 are public? More specifically, are the direct links to the objects with an s3 url accessible by the public?
2. Are s3 buckets indexed by search engines?
3. More broadly, assuming the object urls themselves are not linked in any indexable web pages I'm creating, does making a storage bucket public allow anyone on the internet to easily find the file itself?

**Comment by GaryAustin1** [comment, marked as answer]
1) The public URL, if given out, is and will always work if file is there. 
2) Only if a public URL is used in a website that gets indexed.
3) If you use UUID for files names (or part of) then pretty much someone must have the link to get the file.  IF your RLS is still set correctly on the bucket to not allow listing the files and the like.

**Reply by wkev-n** [reply]
great - thanks!
